devito/types/dimension.py

```diff
@@ -262,7 +262,20 @@ def _arg_values(self, args, interval, grid, **kwargs):
             except (AttributeError, TypeError):
                 pass
 
-        return {self.min_name: loc_minv, self.max_name: loc_maxv}
+        args = {self.min_name: loc_minv, self.max_name: loc_maxv}
+
+        # Maybe override spacing
+        if grid is not None:
+            try:
+                spacing_map = {k.name: v for k, v in grid.spacing_map.items()}
+                args[self.spacing.name] = spacing_map[self.spacing.name]
```

Comment:
why not just 
`args[self.spacing.name] = grid.spacing_map[self.spacing]`

Reply:
see issue #1524 

bottom line: I tried but it breaks and I couldn't come up with a quick patch

Reply:
`args[self.spacing.root.name] = grid.spacing_map[self.spacing.root]` this should always be valid no?  The case where spacing is an expression rather than a symbol should in principle always link to the parent.

Reply:
this one could work, but it still "feels weird" that we have to pick the root in a superclass method

Reply:
with this patch:

```
-                spacing_map = {k.name: v for k, v in grid.spacing_map.items()}
-                args[self.spacing.name] = spacing_map[self.spacing.name]
+                args[self.root.spacing.name] = grid.spacing_map[self.root.spacing]
```

the new test fails with:

```
...

        # The h_x that was passed to the C code must be the one `grid2`, not `grid`
>       assert u2.data[2, 2] == grid2.spacing[0]
E       assert 1.0 == 2.25

test_operator.py:1198: AssertionError
...
```

Reply:
Ooohhhhh, ok now I see it
- `args[self.spacing.name] ` this one is always correct, that's the one to be replaced `self.root.spacing.name` is not in `args`
- Then the `spacing_map` part is needed because while the new `grid` may have the same spacing name `self.spacing` won't be recognized as a key for `grid.spacing_map`.

But now I'm wondering, what if the original gird is `h_x, h_y` and then the new grid is `d_x, d_y`  then `spacing_map['h_x']` wont work with `a `grid2` witch custom dimensions and spacing.
This feels like we need first a dimension ordering in grid, then check the same number of dim then replace spacing by order.

Reply:
for the record:
as discussed offline, this is a bit tricky to do, and since virtually nobody cares about this feature, we decided to proceed with merging